### ISSUE TITLE
go/profiler fails silence when string does not match regex

### ISSUE THREAD

**Issue by johnbalvin** [GoogleCloudPlatform/google-cloud-go#1176]
Code:
```
if err := profiler.Start(profiler.Config{
		Service:   "miPrueba",
	}); err != nil {
		log.Println( err)
	}
```
I was just starting with profiler and was hard to debug why no data was shown and at the end reading carefully I read this from docs:  https://godoc.org/cloud.google.com/go/profiler

> Service must be provided to start the profiler. It specifies the name of
>     // the service under which the profiled data will be recorded and exposed at
>     // the Profiler UI for the project. You can specify an arbitrary string, but
>     // see Deployment.target at
>     // https://github.com/googleapis/googleapis/blob/master/google/devtools/cloudprofiler/v2/profiler.proto
>     // for restrictions

from https://github.com/googleapis/googleapis/blob/master/google/devtools/cloudprofiler/v2/profiler.proto:
`Validation regex for label names: `^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?$``
Where regex is `^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?$` , in short this does not allow capital letter, so this code should return an error but it doesn't, it just fails silence

**Comment by aalexand** [comment]
@nolanmar511 @jqll This configuration error with the agents (using an uppercase letter in the service name) is so common that I suggest we indeed add client-side validation of it. WDYT?

**Comment by nolanmar511** [comment]
I agree that that sounds like a good idea. 
I created a matching issue for the Node.js profiling agent (https://github.com/googleapis/cloud-profiler-nodejs/issues/309)

**Comment by nolanmar511** [comment]
Fixed with https://code-review.googlesource.com/c/gocloud/+/34450

**Comment by nolanmar511** [comment]
Fixed with https://code-review.googlesource.com/c/gocloud/+/34450

**Comment by enocom** [comment]
This was fixed in 764643fb5d9749234b999de50641e8d57d20c9b4. Closing.
